api/src/json-to-codec.ts

```diff
@@ -0,0 +1,37 @@
+import { bytes, codec, context } from "@typeberry/block";
+import type { FastifyReply, FastifyRequest } from "fastify";
+import { createErrorResponse } from "./error";
+import { kinds } from "./kinds";
+
+type Req = { Params: { id: string }; Body: { json: string } };
+type SuccessRes = { data: { codec: string } };
+type Res = { Body: SuccessRes };
+
+export function jsonToCodec(req: FastifyRequest<Req>, reply: FastifyReply<Res>) {
+  const descriptorId = req.params.id;
+  const descriptor = kinds.find((x) => x.name === descriptorId)?.clazz;
+  if (!descriptor) {
+    reply.status(400).send(new Error("Incorrect descriptorId!"));
+    return;
+  }
+
+  const objectToEncode = JSON.parse(req.body.json, (_key, value) => {
+    if (typeof value === "string" && value.startsWith("0x")) {
+      return bytes.Bytes.parseBytes(value, value.length / 2 - 1);
+    }
+    return value;
+  });
```

Comment:
I guess we might need to figure out how to:
1. handle big ints here?
2. Detect `BytesBlobs` (i.e. lengh-prefixed) instead of `Bytes` (fixed-size)

Reply:
1. `isString(x) && new BigInt(x).to.String() === x` ?
2. I have no idea how to distinguish those cases

Reply:
So actually, thinking a bit more about that:
1. Since numbers are just encoded as-is (i.e. not as strings) we won't be able to parse `bigint` correctly, since `value` will already be a `number`. I guess we need to just 🤞 or as you suggested - allow passing numbers as strings.

3. Perhaps this isn't an issue? The codec _might_ work fine if we use `Bytes` everywhere, since it's the descriptor that will know if it should be treated as `BytesBlob` or `Bytes`.